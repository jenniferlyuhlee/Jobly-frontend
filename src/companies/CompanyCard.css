.CompanyCard{
    padding: 1.5rem;
    margin: 2rem 0;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
    text-decoration: none;
}

.CompanyCard:hover{
    color: rgb(94, 184, 219);
}

.card-title{
    margin: 0;
}

.card-body{
    padding: 0.5rem 1rem 0px 1rem;
    display: flex;
    justify-content: space-between;
}

.card-body p{
    font-size: 0.9rem;
}

.card-body img{
    width: 75px;
}