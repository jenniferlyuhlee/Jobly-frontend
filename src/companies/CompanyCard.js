import React from "react";
import { Link } from "react-router-dom";
import "./CompanyCard.css"

function CompanyCard ({company}) {
    return(
        <Link to={`/companies/${company.handle}`} className="CompanyCard card">
            <div>
                <h6 className="card-title">{company.name}</h6>
                <div className="card-body">
                    <p>{company.description}</p>
                    <img src={company.logoUrl}
                        alt={company.name}/>
                </div>
            </div>
        </Link>
    )
}

export default CompanyCard;