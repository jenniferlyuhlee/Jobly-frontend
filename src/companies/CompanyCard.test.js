import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router-dom";

const company1 = {
    name: "Test Company1",
    handle: "test-company1",
    description: "This is a test for company1.",
}

const company2 = {
    name: "Test Company2",
    handle: "test-company2",
    description: "This is a test for company2.",
    logo_url: "https://upload.wikimedia.org/wikipedia/commons/c/ca/LinkedIn_logo_initials.png"
}

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <CompanyCard company={company1}/>
        </MemoryRouter>
    )
});

it("matches snapshot with logo_url", function () {
    const { asFragment } = render(
        <MemoryRouter>
          <CompanyCard company={company1}/>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot with logo_url", function () {
    const { asFragment } = render(
        <MemoryRouter>
          <CompanyCard company={company2}/>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});