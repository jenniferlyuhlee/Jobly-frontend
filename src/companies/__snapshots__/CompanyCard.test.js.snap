// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo_url 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard card"
    href="/companies/test-company1"
  >
    <div>
      <h6
        class="card-title"
      >
        Test Company1
      </h6>
      <div
        class="card-body"
      >
        <p>
          This is a test for company1.
        </p>
        <img
          alt="Test Company1"
        />
      </div>
    </div>
  </a>
</DocumentFragment>
`;

exports[`matches snapshot with logo_url 2`] = `
<DocumentFragment>
  <a
    class="CompanyCard card"
    href="/companies/test-company2"
  >
    <div>
      <h6
        class="card-title"
      >
        Test Company2
      </h6>
      <div
        class="card-body"
      >
        <p>
          This is a test for company2.
        </p>
        <img
          alt="Test Company2"
        />
      </div>
    </div>
  </a>
</DocumentFragment>
`;
